R/azure.R

```diff
@@ -0,0 +1,146 @@
+#' Download specified blobs from Blob Storage and save them in a local dir
+#'
+#' Note that I think it might be wise to instead specify a blob prefix, list the
+#' blobs, and download all the listed blobs. This would let us have some more
+#' flexibility with downloading whole remote directories (like delta tables)
+#'
+#' @param blob_names A vector of blobs to donwload from `container_name`
+#' @param local_dest The path to the local directory to save the files in
+#' @param container_name The Azure Blob Storage container with `blob_names`
+#'
+#' @return NULL on success
+#' @export
+download_from_azure_blob <- function(blob_names, local_dest, container_name) {
+  # Attempt to connect to the storage container
+  blob_container <- rlang::try_fetch(
+    fetch_blob_container(container_name),
+    error = function(con) {
+      cli::cli_abort(
+        c(
+          "Unable to authenticate connection to Blob endpoint",
+          "!" = "Check correct credentials are present as env variables",
+          "!" = "Check container {.var {container_name}} is correct"
+        ),
+        parent = con
+      )
+    }
+  )
+
+  # Attempt to save each blob into local storage
+  for (blob in blob_names) {
+    local_file_path <- file.path(local_dest, blob)
+    rlang::try_fetch(
+      download_file_from_container(
+        blob,
+        blob_container,
+        local_file_path
+      ),
+      error = function(con) {
+        cli::cli_abort(
+          c(
+            "Error downloading blob {.path {blob}}",
+            "Using container {.path {container_name}}",
+            "Writing to local file path {.path local_file_path}"
+          ),
+          parent = con
+        )
+      }
+    )
+  }
+  cli::cli_alert_success("Blobs {.path {blob_names}} downloaded successfully")
+  invisible(NULL)
+}
+
+download_file_from_container <- function(
+    blob_storage_path,
+    container,
+    local_file_path) {
+  cli::cli_alert_info(
+    "Downloading blob {.path {blob_storage_path}} to {.path {local_file_path}}"
+  )
+
+  AzureStor::download_blob(
+    container = container,
+    src = blob_storage_path,
+    dest = local_file_path,
+    overwrite = TRUE
+  )
+
+  cli::cli_alert_success(
+    "Blob {.path {blob_storage_path}} downloaded successfully"
+  )
+
+  invisible(local_file_path)
+}
+
+#' Load Azure Blob endpoint using credentials in environment variables
+#'
+#' This **impure** function depends on the environment variables:
+#' * TENANT_ID
```

Comment:
```suggestion
#' * az_tenant_id
#' * az_subscription_id
#' * az_resource_group
#' * az_storage_account
```

Suggested change to match https://github.com/CDCgov/cfazuR/tree/kg-expand-readme